internal/provider/namespace_export_sink_resource.go

```diff
@@ -0,0 +1,552 @@
+	_ resource.ResourceWithConfigure   = (*namespaceExportSinkResource)(nil)
+	_ resource.ResourceWithImportState = (*namespaceExportSinkResource)(nil)
+
+	namespaceExportSinkSpecModelAttrTypes = map[string]attr.Type{
+		"name":    types.StringType,
+		"enabled": types.BoolType,
+		"s3": types.ObjectType{
+			AttrTypes: s3SpecModelAttrTypes,
+		},
+		"gcs": types.ObjectType{
+			AttrTypes: gcsSpecModelAttrTypes,
+		},
+	}
+
+	s3SpecModelAttrTypes = map[string]attr.Type{
+		"role_name":      types.StringType,
+		"bucket_name":    types.StringType,
+		"region":         types.StringType,
+		"kms_arn":        types.StringType,
+		"aws_account_id": types.StringType,
+	}
+
+	gcsSpecModelAttrTypes = map[string]attr.Type{
+		"sa_id":          types.StringType,
+		"bucket_name":    types.StringType,
+		"gcp_project_id": types.StringType,
+		"region":         types.StringType,
+	}
+)
+
+func (r *namespaceExportSinkResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*client.Client)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected *client.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+
+	r.client = client
+}
+
+func (r *namespaceExportSinkResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_namespace_export_sink"
+}
+
+func (r *namespaceExportSinkResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Provisions a namespace export sink.",
+		Attributes: map[string]schema.Attribute{
+			"namespace": schema.StringAttribute{
+				Description: "The namespace under which the sink is configured. It's needed to be in the format of <namespace>.<account_id>",
+				Required:    true,
+			},
+			"id": schema.StringAttribute{
+				Description: "The unique identifier of the namespace export sink.",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"spec": schema.SingleNestedAttribute{
+				Description: "The specification for the export sink.",
+				Required:    true,
+				Attributes: map[string]schema.Attribute{
+					"name": schema.StringAttribute{
+						Description: "The unique name of the export sink, it can't be changed once set.",
+						Required:    true,
+						PlanModifiers: []planmodifier.String{
+							stringplanmodifier.RequiresReplace(),
+						},
+					},
+					"enabled": schema.BoolAttribute{
+						Description: "A flag indicating whether the export sink is enabled or not.",
+						Computed:    true,
+						Default:     booldefault.StaticBool(true),
+						Optional:    true,
+					},
+					"s3": schema.SingleNestedAttribute{
```

Comment:
If the user can only set one of `s3` or `gcs` can we use the [ExactlyOneOf](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework-validators@v0.17.0/objectvalidator#ExactlyOneOf) validator?